functions/firestore/ItemHelper.js

```diff
@@ -34,11 +35,52 @@ module.exports.removeItem = body => new Promise(async (resolve, reject) => {
   }
 });
 
-module.exports.fetchItemForUser = userId => new Promise(async (resolve, reject) => {
+const fetchItemForUser = userId => new Promise(async (resolve, reject) => {
   try {
     const result = await REFS.COLLECTIONS.ITEMS.where('userId', '==', userId).get();
     resolve(result);
   } catch (error) {
     reject(new Error(error.message));
   }
 });
+
+const closerThanMax = (item) => {
+  return otherItem => {
+    if (!item.location || !otherItem.location) {
+      return false;
+    }
+    return Distance.between(
+      {lat: item.location.lat, lon: item.location.lon},
+      {lat: otherItem.location.lat, lon: otherItem.location.lon}
+    ) < Distance(GEO_DISTANCES.MAX_DISTANCE)
+  };
+};
+
+const findMatchingItems = itemId => new Promise(async (resolve, reject) => {
+  try {
+    const item = (await REFS.COLLECTIONS.ITEMS.doc(itemId).get()).data();
+    const otherType = item.type === 'OFFER' ? 'REQUEST' : 'OFFER';
+    const itemsFromDB = (await REFS.COLLECTIONS.ITEMS
+      .where('type', '==', otherType)
+      .where('categoryId', '==', item.categoryId)
+      .where('status', '==', 'AVAILABLE')
+      .get());
+    if (itemsFromDB.empty) {
+      resolve([]);
+    } else {
+      const itemsToConsider = [];
+      itemsFromDB.forEach((lofasz) => itemsToConsider.push(lofasz.data()));
```

Comment:
lofasz-t azert nevezd at pls, ne menjen be :D

Reply:
Hupsz, az eredetileg csak egy debug valtozo volt 🤣